py/simple_reporter.py

```diff
@@ -135,54 +140,75 @@ def match(file_names, config, quantisation, source, dest_dir):
-        continue
-      
-      #weed out the usable segments and then send them off to the time tiles
-      segments = [ r for r in report['datastore']['reports'] if r['t0'] > 0 and r['t1'] > 0 and r['t1'] - r['t0'] > .5 and r['length'] > 0 and r['queue_length'] >= 0 ]
-      for r in segments:
-        duration = int(round(r['t1'] - r['t0']))
-        start = int(math.floor(r['t0']))
-        end = int(math.ceil(r['t1']))
-        min_bucket = int(start / quantisation)
-        max_bucket = int(end / quantisation)
-        for b in range(min_bucket, max_bucket + 1):
-          tile_level = str(get_tile_level(r['id']))
-          tile_index = str(get_tile_index(r['id']))
-          tile_file_name = dest_dir + os.sep + str(b * quantisation) + '_' + str((b + 1) * quantisation - 1) + os.sep + tile_level + os.sep + tile_index
-          s = [
-            str(r['id']), str(r.get('next_id', INVALID_SEGMENT_ID)), str(duration), '1',
-            str(r['length']), str(r['queue_length']), str(start), str(end), source, 'AUTO'
-          ]
-          tiles.setdefault(tile_file_name, []).append(','.join(s) + os.linesep)
+
+      #find the points where inactivity occurs, these are the window boundaries
+      windows = []
+      for i, point in enumerate(all_points):
+        if i == 0 or point['time'] - points[i - 1]['time'] > inactivity:
+          windows.append(i)
+      windows.append(len(all_points))
+
+      #for each window, the last one is just an end marker
+      for i, window in enumerate(windows, len(windows) - 1):
+        #skip short traces
+        j = windows[i + 1]
+        if j - i < 2:
+          continue
+        
+        #get the matches for it
+        points = all_points[i : j]
+        trace = {'uuid': uuid, 'trace': points}
+        try:
+          match_str = segment_matcher.Match(json.dumps(trace, separators=(',', ':')))
+          match = json.loads(match_str)
+          report = reporter_service.report(match, trace, threshold_sec, report_levels, transition_levels)
+        except (KeyboardInterrupt, SystemExit) as e:
+          raise e
+        except:
+          logger.error('Failed to report trace with uuid %s from file %s' % (uuid, file_name))
+          continue
+        
+        #weed out the usable segments and then send them off to the time tiles
+        buckets = (points[-1]['time'] - points[0]['time']) / quantisation + 1
+        segments = [ r for r in report['datastore']['reports'] if r['t0'] > 0 and r['t1'] > 0 and r['t1'] - r['t0'] > .5 and r['length'] > 0 and r['queue_length'] >= 0 ]
+        for r in segments:
+          duration = int(round(r['t1'] - r['t0']))
+          start = int(math.floor(r['t0']))
+          end = int(math.ceil(r['t1']))
+          min_bucket = int(start / quantisation)
+          max_bucket = int(end / quantisation)
+          diff = max_bucket - min_bucke
+          if diff > buckets:
```

Comment:
except for this line where we actually throw out segments that have crazy times on them. this happens very very rarely but for some reason the start and end times on certain traces' segments have times that are either way in the past or way in the future so we toss those and will fix that problem upstream